core/src/main/java/hudson/model/User.java

```diff
@@ -731,10 +732,43 @@ public synchronized void save() throws IOException, FormValidation {
             throw FormValidation.error(Messages.User_IllegalFullname(fullName));
         }
         if(BulkChange.contains(this))   return;
-        getConfigFile().write(this);
+        synchronized (saving) {
+            saving.add(this);
+        }
+        try {
+            getConfigFile().write(this);
```

Comment:
Would it be possible to move the save/replacer logic into a shared Impl class + Interface with default methods? It seems to be a common use-case for Jenkins which could be reused in other model objects. Having it in external class may help, especially when/if somebody starts working on pluggable storage stories in the core